src/AdhocExploration/DotnetEventProcessor/WarmStorage.EventProcessor/BuildingLookupService.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Newtonsoft.Json;
+
+namespace Microsoft.Practices.IoTJourney.WarmStorage
+{
+    public class BuildingLookupService : IBuildingLookupService
+    {
+        private static ConcurrentDictionary<string, string> cachedBuildingDictionary;
+        private static string blobETag;
+        private static CloudBlockBlob blockBlob;
+        private static bool isGettingLatestETag, isLoadingDictionary;
+        private static Task gettingLatestETagTask, loadingTask;
+        private static DateTimeOffset cacheDateTimeOffset;
+        private static Configuration configuration;
+
+        static BuildingLookupService()
+        {
+            configuration = Configuration.GetCurrentConfiguration();
+            CheckForUpdateAsync().Wait();
```

Comment:
I'm not comfortable doing I/O (or anything async really) in the constructor. I know that we want the constructor to return a ready-to-go service, but in this case I think that we might want a factory or an initialize method.